src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts

```diff
@@ -0,0 +1,202 @@
+                                body: course,
+                                status: 200,
+                            }),
+                        );
+                    },
+                }),
+                MockPipe(ArtemisDurationFromSecondsPipe),
+                MockProvider(JhiAlertService),
+                MockDirective(JhiTranslateDirective),
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        params: {
+                            subscribe: (fn: (value: Params) => void) =>
+                                fn({
+                                    courseId: 1,
+                                }),
+                        },
+                        data: {
+                            subscribe: (fn: (value: any) => void) =>
+                                fn({
+                                    studentExam,
+                                }),
+                        },
+                        snapshot: {
+                            paramMap: convertToParamMap({
+                                courseId: '1',
+                            }),
+                        },
+                    },
+                },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                studentExamDetailComponentFixture = TestBed.createComponent(StudentExamDetailComponent);
+                studentExamDetailComponent = studentExamDetailComponentFixture.componentInstance;
+            });
+    });
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    it('initialize', () => {
+        const courseManagementService = TestBed.inject(CourseManagementService);
+
+        const findCourseSpy = sinon.spy(courseManagementService, 'find');
+        studentExamDetailComponentFixture.detectChanges();
+
+        expect(findCourseSpy).to.have.been.calledOnce;
+        expect(course.id).to.equal(1);
+        expect(studentExamDetailComponent.workingTimeForm).to.not.be.null;
+        expect(studentExamDetailComponent.achievedTotalScore).to.equal(40);
+    });
+
+    it('should return the right icon based on exercise type', () => {
+        exercise = new ModelingExercise(UMLDiagramType.ClassDiagram, course, exerciseGroup);
+        expect(studentExamDetailComponent.exerciseIcon(exercise)).to.equal('project-diagram');
+
+        exercise = new ProgrammingExercise(course, exerciseGroup);
+        expect(studentExamDetailComponent.exerciseIcon(exercise)).to.equal('keyboard');
+
+        exercise = new QuizExercise(course, exerciseGroup);
+        expect(studentExamDetailComponent.exerciseIcon(exercise)).to.equal('check-double');
+
+        exercise = new FileUploadExercise(course, exerciseGroup);
+        expect(studentExamDetailComponent.exerciseIcon(exercise)).to.equal('file-upload');
+    });
+
+    it('should save working time', () => {
+        const studentExamService = TestBed.inject(StudentExamService);
```

Comment:
I'm not sure but is it best practice to move this line into `beforeEach`?

Reply:
I think you right, it is better. Did the changes